gcloud/datastore/helpers.py

```diff
@@ -45,12 +46,17 @@ def entity_from_protobuf(pb):
     :returns: The entity derived from the protobuf.
     """
     key = key_from_protobuf(pb.key)
-    entity = Entity(key=key)
+    entity_props = {}
+    exclude_from_indexes = []
 
     for property_pb in pb.property:
         value = _get_value_from_property_pb(property_pb)
-        entity[property_pb.name] = value
+        entity_props[property_pb.name] = value
+        if not property_pb.value.indexed:
```

Comment:
I can't imagine a valid usecase for indexing one value in a `list_value` but not the others:  we can just document that we assume all items in the list have the same policy, and use the first one to set / remove the field from the entity's list.